messaging/src/main/java/org/springframework/security/messaging/web/csrf/XorCsrfTokenUtils.java

```diff
@@ -56,13 +56,16 @@ static String getTokenValue(String actualToken, String token) {
 		System.arraycopy(actualBytes, randomBytesSize, xoredCsrf, 0, tokenSize);
 
 		byte[] csrfBytes = xorCsrf(randomBytes, xoredCsrf);
-		return Utf8.decode(csrfBytes);
+		return (csrfBytes != null) ? Utf8.decode(csrfBytes) : null;
 	}
 
-	private static byte[] xorCsrf(byte[] randomBytes, byte[] csrfBytes) {
+	static byte[] xorCsrf(byte[] randomBytes, byte[] csrfBytes) {
+		if (csrfBytes.length < randomBytes.length) {
```

Comment:
don't you want to assert that these values equals?
if the `randomBytes` will be larger there is no downside but if the `csrfBytes` is longer then there might be a condition where some bytes of the csrf token is not xored

Reply:
Thanks for your comments on this PR @roysav13! I will be addressing this comment in a separate fix. See gh-15184.